packages/website/src/components/editor/Editor.tsx

```diff
@@ -0,0 +1,357 @@
+import React from 'react';
+
+import { sandboxSingleton } from '../lib/load-sandbox';
+import { createProvideCodeActions } from '../lib/action';
+import { createURI, messageToMarker } from '../lib/utils';
+import { debounce } from '../lib/debounce';
+
+import type {
+  createTypeScriptSandbox,
+  SandboxConfig,
+} from '../../vendor/sandbox';
+import type Monaco from 'monaco-editor';
+import type {
+  WebLinter,
+  LintMessage,
+  TSESTree,
+} from '@typescript-eslint/website-eslint';
+import type { ConfigModel, RuleDetails } from '../types';
+
+interface EditorProps extends ConfigModel {
+  readonly darkTheme: boolean;
+  readonly decoration?: TSESTree.Node | null;
+  readonly onChange?: (
+    value: string,
+    event: Monaco.editor.IModelContentChangedEvent,
+  ) => void;
+  readonly onASTChange?: (
+    value: string | TSESTree.Program,
+    position: Monaco.Position | null,
+  ) => void;
+  readonly onLoadRule?: (value: RuleDetails[]) => void;
+  readonly onSelect?: (position: Monaco.Position) => void;
+  readonly onLoaded?: (tsVersions: readonly string[]) => void;
+}
+
+function shallowEqual(
+  object1: Record<string, unknown> | undefined,
+  object2: Record<string, unknown> | undefined,
+): boolean {
+  if (object1 === object2) {
+    return true;
+  }
+  const keys1 = Object.keys(object1 ?? {});
+  const keys2 = Object.keys(object2 ?? {});
+  if (keys1.length !== keys2.length) {
+    return false;
+  }
+  for (const key of keys1) {
+    if (object1![key] !== object2![key]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+class Editor extends React.Component<EditorProps> {
```

Comment:
I've had good experiences with https://github.com/react-monaco-editor/react-monaco-editor in the past. Any reason do recreate it here?